### PR TITLE
feat: upgrade to HiGlass 1.12.2

### PR DESCRIPTION
**Pull request by sehilyi**
Toward https://github.com/gosling-lang/gos/pull/122

## Change List
 - Upgrade to latest HiGlass (v1.12.2)
 - Make changes regarding the use of the latest d3-related APIs to reflect the breaking changes introduced in the recent HiGlass (e.g., https://github.com/higlass/higlass/pull/1111)
    - The circular brushes were not working in https://github.com/gosling-lang/gos/pull/122 because we are using HiGlass' d3 library (`HGC.libraries.d3Selection.event.sourceEvent`) for the circular brushes, but we were using a syntax that works with previous d3 version.

## Checklist
 - [ ] Ensure the PR works with all demos on the online editor
 - [ ] Unit tests added or updated
 - [ ] Examples added or updated
 - [ ] [Documentation](https://github.com/gosling-lang/gosling-website) updated (e.g., added API functions)
 - [ ] Screenshots for visual changes (e.g., new encoding support or UI change on Editor)

### CODE REVIEW

File: src/gosling-brush/linear-brush-model.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class LinearBrushModel {
     private data: LinearBrushData;
 
     /* drag */
-    private startEvent: typeof D3Selection.event;
+    private startEvent: any; // D3Drag.D3DragEvent.sourceEvent which is `any`
```
Comment (manzt) [inline src/gosling-brush/linear-brush-model.ts]: could we just make this explicit?

```suggestion
    private startEvent: D3Drag.D3DragEvent.sourceEvent;
```

Although it is `any`, if it were to ever change we would automatically get the correct type.

**Review by manzt — APPROVED** [review]
Looks great! Just a minor comment.

**Comment by manzt** [comment]
Do you think we will be able to make a release soon so we can also upgrade gos?

**Comment by sehilyi** [comment]
I received a report that there is an error with `Buffer` when using bigwig files (https://github.com/gosling-lang/gosling.js/issues/876). I think we can release a version once we address it.

**Pull request merged**